internal/apis/certmanager/validation/certificate.go

```diff
@@ -149,6 +149,16 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 		}
 	}
 
+	if crt.NameConstraints != nil {
+		if !crt.IsCA {
```

Comment:
I think we should also add a feature gate check here.

Reply:
The feature gate is at controller level and this validation is at webhook. Is there anything we can do to check the feature gate on controller from here. Introducing a feature gate on webhook help us nothing unless we make sure that both the controller and webhook have the same env vars. 

I hope I am making sense. Please correct me if not.

Reply:
Both the webhook and controller must have feature gating - the webhook must have it to prevent people creating resources which won’t be valid due to the gate being disable. & yep, we expect users to configure the feature gate flag for both the webhook and the controller (similar to say, kube-apiserver and kube-controller-manager)

Reply:
Thanks @munnerz for clarification. I have added the gate in webhook too. PTAL.